apps/native/app/src/screens/home/home.tsx

```diff
@@ -102,16 +128,107 @@ export const MainHomeScreen: NavigationFunctionComponent = ({
   const [refetching, setRefetching] = useState(false)
   const flatListRef = useRef<FlatList>(null)
   const ui = useUiStore()
+  const {
+    homeScreenEnableVehiclesWidget,
+    homeScreenEnableAirDiscountWidget,
+    homeScreenEnableApplicationsWidget,
+    homeScreenEnableInboxWidget,
+    homeScreenEnableLicensesWidget,
+    homeScreenWidgetsInitialised,
+  } = usePreferencesStore()
+
+  const applicationsRes = useListApplicationsQuery({
+    skip: !homeScreenEnableApplicationsWidget,
+  })
 
-  const applicationsRes = useListApplicationsQuery()
   const inboxRes = useListDocumentsQuery({
-    variables: { input: { page: 1, pageSize: 3 } },
+    variables: {
+      input: { page: 1, pageSize: 3 },
+    },
+    skip: !homeScreenEnableInboxWidget,
+  })
+
+  const licensesRes = useGetLicensesData({
+    skipFetching: !homeScreenEnableLicensesWidget,
+  })
+
+  const airDiscountRes = useGetAirDiscountQuery({
+    fetchPolicy: 'network-only',
+    skip: !homeScreenEnableAirDiscountWidget,
+  })
+
+  const vehiclesRes = useListVehiclesQuery({
+    variables: {
+      input: {
+        page: 1,
+        pageSize: 15,
+        showDeregeristered: false,
+        showHistory: false,
+      },
+    },
+    skip: !homeScreenEnableVehiclesWidget,
   })
 
+  useEffect(() => {
+    // If widgets have not been initialized, validate data and set state accordingly
+    if (!homeScreenWidgetsInitialised) {
+      const shouldShowInboxWidget = validateInboxInitialData({ ...inboxRes })
+
+      const shouldShowLicensesWidget = validateLicensesInitialData({
+        ...licensesRes,
+      })
+
+      const shouldShowApplicationsWidget = validateApplicationsInitialData({
+        ...applicationsRes,
+      })
+
+      const shouldShowVehiclesWidget = validateVehiclesInitialData({
+        ...vehiclesRes,
+      })
+
+      const shouldShowAirDiscountWidget = validateAirDiscountInitialData({
+        ...airDiscountRes,
+      })
+
+      preferencesStore.setState({
+        homeScreenEnableInboxWidget: shouldShowInboxWidget,
+        homeScreenEnableLicensesWidget: shouldShowLicensesWidget,
+        homeScreenEnableApplicationsWidget: shouldShowApplicationsWidget,
+        homeScreenEnableVehiclesWidget: shouldShowVehiclesWidget,
+        homeScreenEnableAirDiscountWidget: shouldShowAirDiscountWidget,
+      })
+
+      // Don't set initialized state if any of the queries are still loading
+      if (
```

Comment:
No I don't think so, since the queries take different amount of time to finish. E.g. the licenses query takes a **really** long time to complete, but we have the data for all the other queries way faster, so we want to show data or hide the widget for the other widgets while we show a loading spinner for the licenses widget.